pkg/registry/policy/poddisruptionbudget/strategy.go

```diff
@@ -167,3 +173,30 @@ func (podDisruptionBudgetStatusStrategy) ValidateUpdate(ctx context.Context, obj
 func (podDisruptionBudgetStatusStrategy) WarningsOnUpdate(ctx context.Context, obj, old runtime.Object) []string {
 	return nil
 }
+
+// dropDisabledFields removes disabled fields from the pod disruption budget spec.
+// This should be called from PrepareForCreate/PrepareForUpdate for all resources containing a pod disruption budget spec.
+func dropDisabledFields(pdbSpec, oldPDBSpec *policy.PodDisruptionBudgetSpec) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.PDBPodHealthyPolicy) {
+		if podHealthyPolicyInUse(oldPDBSpec) {
+			// prevent updates to the field, so invalid values cannot be set when validation is off
+			pdbSpec.PodHealthyPolicy = oldPDBSpec.PodHealthyPolicy
```

Comment:
we need this part in case somebody would do `PodRunning` -> `InvalidValue` after we disable the feature, since it also turns off validations for `podHealthyPolicy` field

Reply:
drop this case, validation runs if the field is non-nil regardless of feature gate